src/main/sensors/acceleration.c

```diff
@@ -78,10 +78,13 @@ void accUpdate(timeUs_t currentTimeUs)
 
     applyAccelerationTrims(accelerationRuntime.accelerationTrims);
 
+    float accAdcSquaredSum = 0.0f;
     for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
         const float val = acc.accADC[axis];
         acc.accADC[axis] = accelerationRuntime.accLpfCutHz ? pt2FilterApply(&accelerationRuntime.accFilter[axis], val) : val;
+        accAdcSquaredSum += sq(acc.accADC[axis]);
     }
+    acc.accMagnitude = sqrtf(accAdcSquaredSum) * acc.dev.acc_1G_rec - 1.0f; // used for disarm on impact detection
```

Comment:
I'm a bit concerned about this. Being in free fall means we measure 0G. But with this line we will have negative magnitude in free fall, which is mathematically impossible.

Reply:
Ah = good point!  Thanks!
If we don't subtract 1G, then obviously 1G is normal, not 0G.
I think that's OK, the thresholds just need to be changed elsewhere, to suit (eg GPS Rescue, isAccHealthy.
Let me think on this a bit.

Reply:
Imagine an accelerometer was on an object on a horizontal rail, and we could run it at constant velocity into a 'stop', causing a reliable 10G max deceleration.

While such a rail is horizontal, the root sum of squares would read 1G, and the impact would read close to 10G by sum of squares, near enough anyway.

And now I wonder, if the rail is vertical, and we drive at constant velocity into the same stop - what then will the max recorded deceleration?  All the time, there is positive 1G until we hit, when we add negative 10G.  I'm not sure if my thinking is right, but it seems likely we would read 9G.  

Conversely, if the rail is pointing into the earth, and we drive the same constant velocity into the same stop, we have +1G until we hit, and then add 10G, so I am thinking we record 11G.

I was hoping to make impacts equal, but I don't think it's possible. 

I think you are right; subtracting 1G only makes the problem worse for the vertical direction.

Reply:
You can subtract 1G from the vertical acceleration but that means you first need to know where "down" is. That's possible using the acceleration rotated into world frame (attitude uses it). But acceleration is not a good measure to detect collisions anyway. For that "jerk" is a way better measure.